plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java

```diff
@@ -400,46 +400,50 @@ public void run() {
 		});
 	}
 
-	private void createLogFunctions(IBrowser browser) {
-		browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_LOG, new JsLogFunction(browser,  MessageType.LOG)); 
-		browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_INFO, new JsLogFunction(browser,  MessageType.INFO));  
-		browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_WARN, new JsLogFunction(browser,  MessageType.WARN));  
-		browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_ERROR, new JsLogFunction(browser,  MessageType.ERROR));  
+	private void createLogFunctions(final IBrowser browser) {
+		if (browser != null && !browser.isDisposed()) { 
+			browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_LOG, new JsLogFunction(browser,  MessageType.LOG)); 
+			browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_INFO, new JsLogFunction(browser,  MessageType.INFO));  
+			browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_WARN, new JsLogFunction(browser,  MessageType.WARN));  
+			browser.registerBrowserFunction(ConsoleLogConstants.BROSERSIM_CONSOLE_ERROR, new JsLogFunction(browser,  MessageType.ERROR));  
+		}
 	}
 	
-	private void overrideJsLogFunctions(IBrowser browser) {
-		browser.execute("(function() {" //$NON-NLS-1$
-				+ "if (window.console) {" //$NON-NLS-1$
-				//  Adding BrowserFunction to the process of logging
-				+ 	"var originalConsole = window.console;" //$NON-NLS-1$
-				+ 	"console = {" //$NON-NLS-1$
-				+		"log: function(message) {" //$NON-NLS-1$
-				+			"originalConsole.log(message);" //$NON-NLS-1$
-				+			ConsoleLogConstants.BROSERSIM_CONSOLE_LOG + "(message);" //$NON-NLS-1$
-				+		"}," //$NON-NLS-1$
-				+		"info: function(message) {" //$NON-NLS-1$
-				+			"originalConsole.info(message);" //$NON-NLS-1$
-				+			ConsoleLogConstants.BROSERSIM_CONSOLE_INFO + "(message);" //$NON-NLS-1$
-				+		"},"  //$NON-NLS-1$
-				+		"warn: function(message) {" //$NON-NLS-1$
-				+			"originalConsole.warn(message);" //$NON-NLS-1$
-				+			ConsoleLogConstants.BROSERSIM_CONSOLE_WARN + "(message);" //$NON-NLS-1$
-				+		"},"   //$NON-NLS-1$
-				+		"error: function(message) {" //$NON-NLS-1$
-				+			"originalConsole.error(message);" //$NON-NLS-1$
-				+			ConsoleLogConstants.BROSERSIM_CONSOLE_ERROR + "(message);" //$NON-NLS-1$
-				+		"},"   //$NON-NLS-1$
-				+		"debug: function(message) {" //$NON-NLS-1$ 
-				+			"console.log(message);" //$NON-NLS-1$ do the same as for 'console.log'
-				+		"}"   //$NON-NLS-1$
-				+ 	"};" //$NON-NLS-1$
-
-				// Overriding window.onerror 
-				+	"window.onerror = function(msg, url, lineNumber) {" //$NON-NLS-1$
-				+		"console.error(msg + ' on line ' + lineNumber + ' for ' + url);"  //$NON-NLS-1$
-				+	"}" //$NON-NLS-1$
-				+ "}" //$NON-NLS-1$
-		+ "})();");	 //$NON-NLS-1$
+	private void overrideJsLogFunctions(final IBrowser browser) {
+		if (browser != null && !browser.isDisposed()) {

```

Comment:
@maxandersen frankly, I do not think it is called when the browser is null or disposed in **this** case, due to the fact that there were no NPE or widget disposed for console.log functions. Adding this checks has nothing in common with the original JBIDE-19091 issue (I just did it while refactoring). As I recall removing listeners is unnecessary - they are removed when a control is disposed automatically. However, I do believe you are right about the PR for CordovaSim RippleInjector where the "widget disposed" was caught - https://github.com/jbosstools/jbosstools-browsersim/pull/51